Gems/Terrain/Code/Source/TerrainRenderer/ClipmapBounds.cpp

```diff
@@ -0,0 +1,252 @@
+            Aabb2i& updateRegion = updateRegions.back();
+            
+            if (updatedCenter.m_x < m_center.m_x)
+            {
+                updateRegion.m_min.m_x = updatedCenter.m_x - m_halfSize;
+                updateRegion.m_max.m_x = updateRegion.m_min.m_x + updateWidth;
+            }
+            else
+            {
+                updateRegion.m_max.m_x = updatedCenter.m_x + m_halfSize;
+                updateRegion.m_min.m_x = updateRegion.m_max.m_x - updateWidth;
+            }
+            updateRegion.m_min.m_y = updatedCenter.m_y - m_halfSize;
+            updateRegion.m_max.m_y = updatedCenter.m_y + m_halfSize;
+        }
+        
+        if (updatedCenter.m_y != m_center.m_y && updateWidth < m_size)
+        {
+            updateRegions.push_back();
+            Aabb2i& updateRegion = updateRegions.back();
+            
+            uint32_t updateHeight = AZStd::GetMin<uint32_t>(abs(updatedCenter.m_y - m_center.m_y), m_size);
+            if (updatedCenter.m_y < m_center.m_y)
+            {
+                updateRegion.m_min.m_y = updatedCenter.m_y - m_halfSize;
+                updateRegion.m_max.m_y = updateRegion.m_min.m_y + updateHeight;
+            }
+            else
+            {
+                updateRegion.m_max.m_y = updatedCenter.m_y + m_halfSize;
+                updateRegion.m_min.m_y = updateRegion.m_max.m_y - updateHeight;
+            }
+
+            if (xDiff < 0)
+            {
+                updateRegion.m_min.m_x = updatedCenter.m_x - m_halfSize + updateWidth;
+                updateRegion.m_max.m_x = updatedCenter.m_x + m_halfSize;
+            }
+            else if (xDiff > 0)
+            {
+                updateRegion.m_min.m_x = updatedCenter.m_x - m_halfSize;
+                updateRegion.m_max.m_x = updatedCenter.m_x + m_halfSize - updateWidth;
+            }
+        }
+
+        if (untouchedRegion)
+        {
+            // Default to the entire area being untouched.
+            AZ::Aabb worldBounds = GetWorldBounds();
+            float maxX = worldBounds.GetMax().GetX();
+            float minX = worldBounds.GetMin().GetX();
+            float maxY = worldBounds.GetMax().GetY();
+            float minY = worldBounds.GetMin().GetY();
+
+            if (updatedCenter.m_x < m_center.m_x)
+            {
+                maxX = (updatedCenter.m_x + m_halfSize) * m_rcpScale;
+            }
+            else if (updatedCenter.m_x > m_center.m_x)
+            {
+                minX = (updatedCenter.m_x - m_halfSize) * m_rcpScale;
+            }
+            if (updatedCenter.m_y < m_center.m_y)
+            {
+                maxY = (updatedCenter.m_y + m_halfSize) * m_rcpScale;
+            }
+            else if (updatedCenter.m_y > m_center.m_y)
+            {
+                minY = (updatedCenter.m_y - m_halfSize) * m_rcpScale;
+            }
+
+            untouchedRegion->Set(AZ::Vector3(minX, minY, 0.0f), AZ::Vector3(maxX, maxY, 0.0f));
```

Comment:
Not sure, but might be useful to set Z to flt_min and flt_max, so that you can use the AABB to compare against any other AABB and automatically get a 2D compare.

Reply:
Hmm... I lean towards leaving this at zero and would just us a 2D aabb if one existed. Its easy enough for the caller to set the min/max z to whatever they want if they want to use it for 3d intersection tests. Changing this would also mess up one of the unit tests that uses the area of the untouched region as part of the test (not super hard to fix, but returning an AABB that would return inf for GetSurfaceArea() doesn't seem ideal).